### PR TITLE
feat: add gov proposal execution for e2e testsuite

### PR DESCRIPTION
**Pull request by colin-axner**
## Description

See #2071 for passing test usage

closes: #2113

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: e2e/testsuite/testsuite.go [hunk 1]
```diff
@@ -374,3 +379,38 @@ func GetNativeChainBalance(ctx context.Context, chain ibc.Chain, user *ibc.Walle
...
+
+	err = chain.VoteOnProposalAllValidators(ctx, "1", ibc.ProposalVoteYes)
+	s.Require().NoError(err)
+
+	// ensure voting period has not passed before validators finished voting
+	proposal, err = s.QueryProposal(ctx, chain, 1)
+	s.Require().NoError(err)
+	s.Require().Equal(govtypes.StatusVotingPeriod, proposal.Status)
+
+	time.Sleep(time.Second * 30) // pass proposal
```
Comment (chatton) [inline e2e/testsuite/testsuite.go:411]: should we maybe wait slightly longer the specified time, maybe we could run into an issue where it is not yet passed.

Or better yet we could wait for the status to equal passed with a timeout of a little over 30 seconds. What do you think?

Comment (colin-axner) [reply e2e/testsuite/testsuite.go:411]: I like your suggestion [here](https://github.com/cosmos/ibc-go/issues/1981#issuecomment-1220482819). Maybe we could fix this in a followup pr? 

I can modify the time to be slightly longer than the voting period, but I figure we will fine tune this functionality over time anyways.

Comment (chatton) [reply e2e/testsuite/testsuite.go:411]: yep that sounds good to be 👍

**Review by damiannolan — APPROVED** [review]
Nice! Will we also need to support the new way of using gov proposal with v0.46?

**Comment by colin-axner** [comment]
> Will we also need to support the new way of using gov proposal with v0.46?

Yea I think so, but those won't be backwards compatible for cross compatibility testing, so we can add that function for future usage when we start migrating our proposals to messages. Essentially we will just have duplicates of legacy and new for a while until we remove legacy as those versions no longer become supported

**Pull request merged**